formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Encoding.kt

```diff
@@ -428,6 +448,64 @@ internal class CborDecoder(private val input: ByteArrayInput) {
         return result
     }
 
+    fun skipElement() {
```

Comment:
This method is quite huge and hard to read.
It would be nice if you could extract a few subroutines (e.g. one from `if (prune) { ... }` can be replaced with `if (prune) prune()`, `length` extraction and so on) to make it as straightforward as possible

Reply:
> This method is quite huge and hard to read.

Agreed.

> `if (prune) { ... }` can be replaced with `if (prune) prune()`

I was thinking that too, but I worried that a `prune()` function is very specific to element skipping and a `prune()` function in the `CborDecoder` class feels like the `CborDecoder` is taking on too many responsibilities? Would it warrant a separate `private` class that is responsible for all this element skipping logic (to help isolate the element skipping functions from other CBOR decoding functionality)? Or is that overkill?

Reply:
IMO It's a bit too much, let's stick with single `CborDecoder ` for now.